data-prepper-plugins/mongodb/src/main/java/org/opensearch/dataprepper/plugins/mongo/export/ExportPartitionWorker.java

```diff
@@ -181,8 +181,8 @@ public void run() {
                         final String primaryKeyBsonType = primaryKeyDoc.map(bsonDocument -> bsonDocument.get(DOCUMENTDB_ID_FIELD_NAME).getBsonType().name()).orElse(UNKNOWN_TYPE);
 
                         // The version number is the export time minus some overlap to ensure new stream events still get priority
-                        final long eventVersionNumber = (exportStartTime - VERSION_OVERLAP_TIME_FOR_EXPORT.toMillis()) * 1_000;
-                        final Event event = recordConverter.convert(record, exportStartTime, eventVersionNumber, primaryKeyBsonType);
+                        final long eventVersionNumber = (exportStartTimeEpochMillis - VERSION_OVERLAP_TIME_FOR_EXPORT.toMillis()) * 1_000L;
```

Comment:
Ques: why times 1000?

Reply:
converting time to epoch nanos to be consistent for stream and export events